cloudinit/config/tests/test_resolv_conf.py

```diff
@@ -0,0 +1,86 @@
+from unittest import mock
+
+import pytest
+
+from cloudinit.config.cc_resolv_conf import generate_resolv_conf
+
+
+EXPECTED_HEADER = """\
+# Your system has been configured with 'manage-resolv-conf' set to true.
+# As a result, cloud-init has written this file with configuration data
+# that it has been provided. Cloud-init, by default, will write this file
+# a single time (PER_ONCE).
+#\n\n"""
+
+
+class TestGenerateResolvConf:
+    @mock.patch("cloudinit.config.cc_resolv_conf.templater.render_to_file")
+    def test_default_target_fname_is_etc_resolvconf(self, m_render_to_file):
+        generate_resolv_conf("templates/resolv.conf.tmpl", mock.MagicMock())
+
+        assert [
+            mock.call(mock.ANY, "/etc/resolv.conf", mock.ANY)
+        ] == m_render_to_file.call_args_list
+
+    @mock.patch("cloudinit.config.cc_resolv_conf.templater.render_to_file")
+    def test_target_fname_is_used_if_passed(self, m_render_to_file):
+        generate_resolv_conf(
+            "templates/resolv.conf.tmpl", mock.MagicMock(), "/use/this/path"
+        )
+
+        assert [
+            mock.call(mock.ANY, "/use/this/path", mock.ANY)
+        ] == m_render_to_file.call_args_list
+
+    # Patch in templater so we can assert on the actual generated content
+    @mock.patch("cloudinit.templater.util.write_file")
+    # Parameterise with the value to be passed to generate_resolv_conf as the
+    # `params` parameter, and a list of the expected lines after the header as
+    # `extra_lines`.
+    @pytest.mark.parametrize(
```

Comment:
parameterize ? or parametrize?

Reply:
both?  wth!

```
>>> pytest.mark.parameterize 
<MarkDecorator 'parameterize' {'args': (), 'kwargs': {}}>
>>> pytest.mark.parametrize 
<MarkDecorator 'parametrize' {'args': (), 'kwargs': {}}>
```

Reply:
[`pytest.mark`](https://docs.pytest.org/en/latest/mark.html) is a generic API for annotating test functions/methods. `parametrize` is one of the builtin markers, alongside `skip`, `skipif` and `xfail`.

If you attempt to run the tests with `pytest.mark.parameterize` instead of `pytest.mark.parametrize`, you get a failure during the collect phase:

```
test_flags_and_options has 'parameterize' mark, spelling should be 'parametrize'
```

Reply:
Interesting.